Add support for running SOF on i.MX93's A55 core

LaurentiuM1234 opened the pull request:
This series of patches introduces all the changes required to make SOF using Zephyr run on i.MX93's A55 core.

LaurentiuM1234 committed:
boards: arm64: mimx93_evk: Introduce new board for SOF usage

This commit introduces a new variant of the mimx93_evk_a55 board
used to run the SOF module.

This is required because we don't want to pollute the imx93_a55
generic board with SOF-specific nodes such as the memory nodes.
Also, the SOF-specific drivers should not be activated in the
generic board.

SOF with Jailhouse also requires some specific configurations
such as the 40-bit PAs and VAs and the static data cache line
size so this is another reason for which this new
board had to be created.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

LaurentiuM1234 committed:
soc: arm64: nxp: mimx9: Modify linker script to include SOF sections

Since SOF uses some extra sections we need to add them to the linker
script. The used linker script is simply a copy of arm64 linker script
with SOF regions included. This is because conditionally adding sections
to the main arm64 linker script is not really desirable.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

LaurentiuM1234 committed:
soc: arm64: nxp: mimx9: Create identity mappings for regions in SOF dts

This commit creates identity mappings for all memory regions required
by SOF. This is needed because drivers using these memory regions have
not yet been ported to Zephyr.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

LaurentiuM1234 commented:
CC: @dbaluta @iuliana-prodan @paulstelian97

paulstelian97 commented on soc/arm64/nxp_imx/mimx9/linker.ld:
```diff
@@ -1,8 +1,354 @@
 /*
- * Copyright 2022 NXP
+ * Copyright (c) 2013-2014 Wind River Systems, Inc.
```
Unless this is essentially a full replacement, removing an old copyright might not be a good idea. I'd say at least a bit of discussion is warranted.

LaurentiuM1234 replied:
It's basically a copy/paste of the arm64's linker script with some SOF additions. I guess due to these additions we can keep the copyright. Will fix.

dbaluta commented on boards/arm64/mimx93_evk/mimx93_evk_a55_sof.dts:
```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 NXP
```
We need more details here in the commit message about why do we need to introduce a new board and what's the general setup of running SOF.

carlocaione reviewed (commented):
Can we have a README for this board and how to use it?

LaurentiuM1234 commented:
> Can we have a README for this board and how to use it?

Sure, would it be alright to also document it in mimx93_evk's index.rst?

carlocaione commented:
> Sure, would it be alright to also document it in mimx93_evk's index.rst?

Yup, sounds fine.

LaurentiuM1234 commented:
@carlocaione Should be a bit better now. Please let me know if there's something unclear in the documentation so I can rectify it.

@dbaluta ACK most of your comments. The Jailhouse-related concepts should be covered in the documentation now.

LaurentiuM1234 committed:
boards: arm64: mimx93_evk: doc: Add information about SOF board variant

This commit adds some additional information about the SOF variant
of the mimx93_evk board. It mostly goes into how SOF with Zephyr
is going to work using this board and some basic Jailhouse-related
concepts.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

povergoing commented on boards/arm64/mimx93_evk/mimx93_evk_a55_sof_defconfig:
```diff
@@ -0,0 +1,44 @@
+CONFIG_BOARD_MIMX93_EVK_A55=y
+
+# Zephyr Kernel Configuration
+CONFIG_XIP=n
```
nit: the default val of  `CONFIG_XIP` is `n`, I think no need to set with n

povergoing commented on boards/arm64/mimx93_evk/mimx93_evk_a55_sof_defconfig:
```diff
@@ -0,0 +1,44 @@
+CONFIG_ICACHE_LINE_SIZE_DETECT=y
+# Currently, SOF doesn't allow for run-time
+# data cache line size detection.
+CONFIG_DCACHE_LINE_SIZE_DETECT=n
```
same with `CONFIG_XIP`, the default of `CONFIG_DCACHE_LINE_SIZE_DETECT` is `n`. But since this CONFIG might be a kind of alert, I am not sticking to it.

LaurentiuM1234 replied:
Fixed. Added a more descriptive message which should itself serve as a warning.
